src/sdam/state/mod.rs

```diff
@@ -47,6 +51,7 @@ pub(crate) struct WeakTopology {
 /// manager and the client options.
 #[derive(Clone, Debug)]
 struct Common {
+    is_alive: Arc<AtomicBool>,
```

Comment:
Even though AtomicBool is atomic, we still need to use an Arc so that changing it on one copy changes it on all the others.